src/hotspot/cpu/aarch64/vm_version_aarch64.cpp

```diff
@@ -564,26 +564,40 @@ void VM_Version::initialize() {
   check_virtualizations();
 }
 
-void VM_Version::check_virtualizations() {
 #if defined(LINUX)
-  const char* info_file = "/sys/devices/virtual/dmi/id/product_name";
-  // check for various strings in the dmi data indicating virtualizations
+static bool check_info_file(const char* fpath,
+                            const char* virt1, VirtualizationType vt1,
+                            const char* virt2, VirtualizationType vt2) {
```

Comment:
We should (future RFE) convert VirtualizationType to a class/struct so that we don't need to manually pair the name with the value.